packages/dashboard/src/app/views/myStories/content/storyGridView/index.js

```diff
@@ -94,19 +90,41 @@ const StoryGridView = ({
       const newFocusId = returnStoryFocusId?.value;
       setActiveGridItemId(newFocusId);
       // grab the menu button and refocus
-      itemRefs.current?.[newFocusId]
-        ?.querySelector(`.${CONTEXT_MENU_BUTTON_CLASS}`)
-        ?.focus();
+      const firstFocusableElement = itemRefs.current?.[
+        newFocusId
+      ]?.querySelectorAll(['button', 'a'])?.[0];
+
+      firstFocusableElement?.focus();
     }
   }, [activeGridItemId, returnStoryFocusId]);
 
-  // when keyboard focus changes through FocusableGridItem immediately focus the edit preview layer on top of preview
+  // when keyboard focus changes through FocusableGridItem
+  // immediately focus the edit preview layer on top of preview
   useEffect(() => {
     if (activeGridItemId) {
       activeGridItemIdRef.current = activeGridItemId;
     }
   }, [activeGridItemId]);
 
+  // Additional functionality needed when closing context menus to maintain grid item focus
+  const handleMenuToggle = useCallback(
+    (evt, id) => {
+      storyMenu.handleMenuToggle(id);
+      // Conditionally return the focus to the grid when menu is closed
+      if (id < 0 && evt?.keyCode === 9) {
```

Comment:
So this only runs when `tab` is used - okay cool! I like the comment but maybe we should add that as a comment here too since `9` isn't super obvious to me

```suggestion
      // Conditionally return the focus to the grid when menu is closed using `tab`
      if (id < 0 && evt?.keyCode === 9) {
```